plugins/out_stackdriver/stackdriver.c

```diff
@@ -2510,6 +2515,91 @@ static void update_retry_metric(struct flb_stackdriver *ctx,
 }
 #endif
 
+static int extract_map_from_msgpack(msgpack_object *root, char *name, int size, msgpack_object_type object_type ,msgpack_object *val){
+    int i;
+    msgpack_object key;
+    for (i = 0; i < root->via.map.size; i++) {
+        key = root->via.map.ptr[i].key;
+        if (key.type != MSGPACK_OBJECT_STR) {
+                continue;
+        }
+        if(key.via.str.size == size && strncmp(key.via.str.ptr, name, size) == 0){
+            *val = root->via.map.ptr[i].val;
+            if (val->type != object_type) {
+                return -1;
+            }
+            return 0;
+        }
+    }
+    return -1;
+}
+
+static int parse_partial_success_response(struct flb_http_client *c, struct flb_stackdriver *ctx, int *partial_failure){
+    int ret;
+    int root_type;
+    int i;
+    char *buffer;
+    size_t size;
+    size_t off = 0;
+    msgpack_unpacked result;
+    msgpack_object root;
+    msgpack_object error_map;
+    msgpack_object details_arr;
+    msgpack_object logEntryErrors_map;
+    ret = flb_pack_json(c->resp.payload, c->resp.payload_size,
+        &buffer, &size, &root_type, NULL);
+    msgpack_unpacked_init(&result);
+    ret = msgpack_unpack_next(&result, buffer, size, &off);
+    if (ret != MSGPACK_UNPACK_SUCCESS) {
+        flb_plg_error(ctx->ins, "Cannot unpack %s response\n%s", c->resp.payload);
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+
+    root = result.data;
+    if (root.type != MSGPACK_OBJECT_MAP) {
+        flb_plg_error(ctx->ins, "%s response parsing failed, msgpack_type=%i", root.type);
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+
+    ret = extract_map_from_msgpack(&root, "error", 5, MSGPACK_OBJECT_MAP ,&error_map);
+    if (ret == -1) {
+        flb_plg_error(ctx->ins, "%s response parsing failed, could not find key: \"error\"");
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+    ret = extract_map_from_msgpack(&error_map, "details", 7, MSGPACK_OBJECT_ARRAY ,&details_arr);
+    if (ret == -1) {
+        flb_plg_error(ctx->ins, "%s response parsing failed, could not find key: \"details\"");
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+    for (i = 0; i < details_arr.via.array.size; i++) {
+        logEntryErrors_map = details_arr.via.array.ptr[i];
+        if (logEntryErrors_map.type != MSGPACK_OBJECT_MAP) {
+            continue;
+        }
+        ret = extract_map_from_msgpack(&logEntryErrors_map, "logEntryErrors", 14, MSGPACK_OBJECT_MAP ,&logEntryErrors_map);
+        if (ret == -1){
+            flb_plg_error(ctx->ins, "%s response parsing failed, could not find key: \"logEntryErrors\"");
+            flb_free(buffer);
+            msgpack_unpacked_destroy(&result);
+            return -1;
+        }
+        break;
```

Comment:
So we're just looking for the first thing in the array that is actually a map? Is that an assumption we can rely on?